core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -100,6 +123,28 @@ public static String propertyAsString(
     return defaultValue;
   }
 
+  /**
+   * Validate the table commit related properties to have non-negative integer on table creation to
+   * prevent commit failure
+   */
+  public static void validateCommitProperties(Map<String, String> properties) {
+    for (String commitProperty : COMMIT_PROPERTIES) {
+      if (properties.containsKey(commitProperty)) {
```

Comment:
how about doing it the way you do in the above method?  I think slightly less calls to properties?

int value = properties.get()
if (value != null) {
  int parsedValue;
  ...
}